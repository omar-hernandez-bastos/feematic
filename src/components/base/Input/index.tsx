import React from 'react'

interface InputProps extends React.HTMLProps<HTMLInputElement> {
  label: string
}

const Input = ({ label, ...rest }: InputProps) => {
  return (
    <div className="relative py-4">
      <label htmlFor="email" className="text-sm font-semibold text-gray-200">
        {label}
      </label>
      <input
        {...rest}
        className="p-4 w-full h-10 placeholder-transparent text-gray-900 bg-gray-100 rounded-xl border border-gray-300 focus:border-cyan-600 focus:outline-none"
      />
    </div>
  )
}
export default Input
