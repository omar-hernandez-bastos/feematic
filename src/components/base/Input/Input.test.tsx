import { render } from '@testing-library/react'
import Input from '.'

describe('Input Test', () => {
  it('Input Component should render correctly', () => {
    const InputProps = { label: 'test' }
    const { container } = render(<Input {...InputProps} />)
    expect(container).toMatchSnapshot()
  })
})
