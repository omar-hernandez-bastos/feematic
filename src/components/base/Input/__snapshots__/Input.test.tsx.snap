// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test Input Component should render correctly 1`] = `
<div>
  <div
    class="relative py-4"
  >
    <label
      class="text-sm font-semibold text-gray-200"
      for="email"
    >
      test
    </label>
    <input
      class="p-4 w-full h-10 placeholder-transparent text-gray-900 bg-gray-100 rounded-xl border border-gray-300 focus:border-cyan-600 focus:outline-none"
    />
  </div>
</div>
`;
